File: frontend/src/api/skillGaps.ts
import { apiFetch } from "./client";

export type GapStatus = "identified" | "learning" | "acquired" | "not_pursuing";
export type GapCategory = "hard_skill" | "tool" | "domain" | "seniority";

export interface SkillGap {
  id: string;
  job_id: string;
  category: GapCategory;
  skill_name: string;
  priority: number;
  why_it_matters: string;
  suggested_resource: string;
  status: GapStatus;
  created_at: string;
}

export const analyzeGaps = (job_id: string, base_resume_id: string) =>
  apiFetch<SkillGap[]>("/skill-gaps/analyze", {
    method: "POST",
    body: JSON.stringify({ job_id, base_resume_id }),
  });

export const listGaps = (job_id?: string) =>
  apiFetch<SkillGap[]>(`/skill-gaps${job_id ? `?job_id=${job_id}` : ""}`);

export const updateGapStatus = (id: string, status: GapStatus) =>
  apiFetch<SkillGap>(`/skill-gaps/${id}`, {
    method: "PATCH",
    body: JSON.stringify({ status }),
  });

File: api/app/schemas/skill_gaps.py
import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class SkillGapAnalyzeRequest(BaseModel):
    job_id: uuid.UUID
    base_resume_id: uuid.UUID


class SkillGapItem(BaseModel):
    id: uuid.UUID
    job_id: uuid.UUID
    category: str
    skill_name: str
    priority: int
    why_it_matters: str
    suggested_resource: str
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}


class SkillGapUpdate(BaseModel):
    status: Literal["identified", "learning", "acquired", "not_pursuing"]
